priorityq.py

```diff
@@ -60,10 +61,25 @@ def __init__(self, iterable=()):
         args:
             iterable: an optional iterable to add to the priority queue. Items
                       added this way will be given a priority of None.
+
+        each item inside iterable can be either:
+        * A QNode object
+        * A container with value, priority
+        * A non-iterable value
+
         """
         self.heap = BinaryHeap(iterable=())
         for item in iterable:
-            self.insert(item)
+            try:
+                is_container = len(item) == 2
```

Comment:
We might want to consider a different method other then `len(item) == 2` to determine if this is a `(val, priority)` tuple here. As is, it will catch cases that we don't want.

For example, if a string of length 2 comes in as an item like `BinaryHeap(['hi', (10, 0), ('something else', 1)])`, it will potentially examine `'hi'`, find it has length 2 and then:

`self.insert('h', 'i')`

Which isn't what we want.